src/ethereum/frontier/state.py

```diff
@@ -0,0 +1,218 @@
+"""
+State
+^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+The state contains all information that is preserved between transactions.
+
+It consists of a main account trie and storage tries for each contract.
+"""
+from dataclasses import dataclass, field
+from typing import Callable, Dict
+
+from ethereum.base_types import U256, Bytes, modify
+
+from .eth_types import EMPTY_ACCOUNT, Account, Address, Root
+from .trie import EMPTY_TRIE_ROOT, Trie, root, trie_get, trie_set
+
+
+@dataclass
+class State:
+    """
+    Contains all information that is preserved between transactions.
+    """
+
+    _main_trie: Trie[Account] = field(
+        default_factory=lambda: Trie(secured=True, default=EMPTY_ACCOUNT)
+    )
+    _storage_tries: Dict[Address, Trie[U256]] = field(default_factory=dict)
+
+
+def get_account(state: State, address: Address) -> Account:
+    """
+    Get the `Account` object at an address. Returns EMPTY_ACCOUNT if there
+    is no account the address.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address to lookup.
+
+    Returns
+    -------
+    account : `Account`
+        Account at address.
+    """
+    account = trie_get(state._main_trie, address)
+    assert isinstance(account, Account)
+    return account
+
+
+def set_account(state: State, address: Address, account: Account) -> None:
+    """
+    Set the `Account` object at an address.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address to set.
+    account : `Account`
+        Account to set at address.
+    """
+    trie_set(state._main_trie, address, account)
```

Comment:
I feel it would be giving more clarity to the readers if we mentioned here and in docstring that if the account is `EMPTY_ACCOUNT`, then the address and account pair would be removed from the state trie. Please let me know what you think. Thanks.